src/auth/webhook-secret.guard.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import * as crypto from 'crypto';
+import { Request } from 'express';
+import { Env } from 'src/env';
+
+@Injectable()
+export class WebhookSecretGuard implements CanActivate {
+  canActivate(context: ExecutionContext): boolean {
+    const request: Request = context.switchToHttp().getRequest();
+    const signature = request.headers['x-hub-signature-256'] as string;
+    const secret = Env.SECRET.WEBHOOK;
+    const payload = JSON.stringify(request.body);
```

Comment:
PullProofAI

There is a potential issue with handling errors during the creation of the HMAC digest.
```typescript
13:   canActivate(context: ExecutionContext): boolean {
14:     const request: Request = context.switchToHttp().getRequest();
15:     const signature = request.headers['x-hub-signature-256'] as string;
16:     const secret = Env.SECRET.WEBHOOK;
17:     const payload = JSON.stringify(request.body);
```

#### Suggested Fix
Ensure proper error handling for the creation of the HMAC digest to prevent unhandled exceptions.
```typescript
13:   canActivate(context: ExecutionContext): boolean {
14:     const request: Request = context.switchToHttp().getRequest();
15:     const signature = request.headers['x-hub-signature-256'] as string;
16:     const secret = Env.SECRET.WEBHOOK;
17:     const payload = JSON.stringify(request.body);

// Add try-catch block for error handling
try {
    const hmac = crypto.createHmac('sha256', secret);
    const digest = `sha256=${hmac.update(payload).digest('hex')}`;

    if (!crypto.timingSafeEqual(Buffer.from(digest), Buffer.from(signature))) {
        throw new UnauthorizedException('Invalid signature');
    }

    return true;
} catch (error) {
    throw new UnauthorizedException('Error creating HMAC digest');
}
```


<!-- This is an auto-generated comment by PullProofAI -->